tycho-surefire/org.eclipse.tycho.surefire.osgibooter/src/main/java/org/eclipse/tycho/surefire/osgibooter/OsgiSurefireBooter.java

```diff
@@ -106,6 +106,7 @@ public static int run(String[] args, Properties testProps) throws Exception {
         Bundle bundle = FrameworkUtil.getBundle(OsgiSurefireBooter.class);
         urls.add(getURL(bundle));
         Bundle[] fragments = getFragments(bundle);
+        Arrays.sort(fragments);
```

Comment:
Instead of sorting, can we better use a `LinkedHashMap` in `getFragments`?
The reason is that sorting would imply some kind of precedence, while all we want here is that the order is the same between two runs.

Reply:
Okay that was misleading.. as you described the Map is not used by Tycho (what would need to be changed in Equinox). But as we already have a stream there, we could simply use 

hostWires.stream().map(wire -> wire.getRequirer().getBundle()).filter(Objects::nonNull)**.sorted()**
                .toArray(Bundle[]::new);

Reply:
I don't know, the information from `getFragments()` comes from other hash maps... I gave up on trying to follow that since the bundle wiring (that is used there) has tons of initializations. If you insist I can continue debugging, it didn't seem trivial to find the initial hash map (and there is at least one more hash map on the way).